include/nlohmann/detail/macro_scope.hpp

```diff
@@ -97,6 +97,20 @@
     #define JSON_HAS_FILESYSTEM 0
 #endif
 
+#ifndef JSON_HAS_THREE_WAY_COMPARISON
+    #if defined(__cpp_lib_three_way_comparison) && __cpp_lib_three_way_comparison >= 201907L \
+        && defined(__cpp_impl_three_way_comparison)&& __cpp_impl_three_way_comparison >= 201907L
+        #define JSON_HAS_THREE_WAY_COMPARISON
+    #endif
+#endif
+
+#ifdef JSON_HAS_THREE_WAY_COMPARISON
+    #undef JSON_HAS_THREE_WAY_COMPARISON
+    #define JSON_HAS_THREE_WAY_COMPARISON 1
```

Comment:
Why not just do the `#define JSON_HAS_THREE_WAY_COMPARISON 1` above?

Reply:
To cover the case when it has been externally defined for whatever reason.
Following the pattern established here:
```
// C++ language standard detection
// if the user manually specified the used c++ version this is skipped
#if !defined(JSON_HAS_CPP_20) && !defined(JSON_HAS_CPP_17) && !defined(JSON_HAS_CPP_14) && !defined(JSON_HAS_CPP_11)
```

I plan on changing all boolean macros to be defined as 0 or 1 eventually (useful for an extension to part 2 of this PR).

Reply:
In those other cases, if it has been externally defined, we just leave it as-is.  We don't blindly change it to `1`.  This will break if someone has `#define JSON_HAS_THREE_WAY_COMPARISON 0`.

Reply:
You're right.
My end goal is to be able to `static_cast<bool>(ANY_BOOLEAN_MACRO)`, expose that through `basic_json<>::meta()`, and output it as part of the build information (see part 2 of this PR).
I was trying to avoid 
```
#ifdef BOOLEAN_MACRO
meta["BOOLEAN_MACRO"] = true;
#else
meta["BOOLEAN_MACRO"] = false;
#endif
```
and just write
```
meta["BOOLEAN_MACRO"] = static_cast<bool>(BOOLEAN_MACRO);
```

Reply:
Makes sense, so you just need this:

    #ifndef JSON_HAS_THREE_WAY_COMPARISON
        #if defined(__cpp_lib_three_way_comparison) && __cpp_lib_three_way_comparison >= 201907L \
            && defined(__cpp_impl_three_way_comparison)&& __cpp_impl_three_way_comparison >= 201907L
            #define JSON_HAS_THREE_WAY_COMPARISON 1
        #else
            #define JSON_HAS_THREE_WAY_COMPARISON 0
        #endif
    #endif

Reply:
Done.